packages/compiler/test/decorators/examples.test.ts

```diff
@@ -0,0 +1,361 @@
+import { ok } from "assert";
+import { describe, expect, it } from "vitest";
+import { Operation, getExamples, getOpExamples, serializeValueAsJson } from "../../src/index.js";
+import { expectDiagnostics } from "../../src/testing/expect.js";
+import { createTestRunner } from "../../src/testing/test-host.js";
+
+async function getExamplesFor(code: string) {
+  const runner = await createTestRunner();
+  const { test } = await runner.compile(code);
+
+  ok(test, "Expect to have @test type named test.");
+  return {
+    program: runner.program,
+    target: test,
+    examples: getExamples(runner.program, test as any),
+  };
+}
+
+async function getOpExamplesFor(code: string) {
+  const runner = await createTestRunner();
+  const { test } = (await runner.compile(code)) as { test: Operation };
+
+  ok(test, "Expect to have @test type named test.");
+  return {
+    program: runner.program,
+    target: test,
+    examples: getOpExamples(runner.program, test as any),
+  };
+}
+
+async function diagnoseCode(code: string) {
+  const runner = await createTestRunner();
+  return await runner.diagnose(code);
+}
+
+describe("@example", () => {
+  describe("model", () => {
+    it("valid", async () => {
+      const { program, examples, target } = await getExamplesFor(`
+      @example(#{ a: 1, b: 2 })
+      @test model test {
+        a: int32;
+        b: int32;
+      }
+    `);
+      expect(examples).toHaveLength(1);
+      expect(serializeValueAsJson(program, examples[0].value, target)).toEqual({ a: 1, b: 2 });
+    });
+
+    it("emit diagnostic for missing property", async () => {
+      const diagnostics = await diagnoseCode(`
+        @example(#{ a: 1 })
+        @test model test {
+          a: int32;
+          b: int32;
+        }
+      `);
+      expectDiagnostics(diagnostics, {
+        code: "missing-property",
+      });
+    });
+  });
+
+  describe("model property", () => {
+    it("valid", async () => {
+      const { program, examples, target } = await getExamplesFor(`
+      model TestModel {
+        @example(1)
+        @test test: int32;
+        b: int32;
+      }
+    `);
+      expect(examples).toHaveLength(1);
+      expect(serializeValueAsJson(program, examples[0].value, target)).toEqual(1);
+    });
+
+    it("emit diagnostic for unassignable value", async () => {
+      const diagnostics = await diagnoseCode(`
+        model TestModel {
+          @example("abc")
+          @test test: int32;
+          b: int32;
+        }
+      `);
+      expectDiagnostics(diagnostics, {
+        code: "unassignable",
+      });
+    });
+  });
+
+  describe("scalar", () => {
+    it("valid", async () => {
+      const { program, examples, target } = await getExamplesFor(`
+      @example(test.fromISO("11:32"))
+      @test scalar test extends utcDateTime;
```

Comment:
should have an example of each valid scalar type

Reply:
there is test cases below that go over serialization for each scalar type, this was just to test you can apply to a scalar